src/native/corehost/fxr/sdk_resolver.cpp

```diff
@@ -237,14 +225,27 @@ sdk_resolver sdk_resolver::from_nearest_global_file(const pal::string_t& cwd, bo
 {
     sdk_resolver resolver{ allow_prerelease };
 
-    if (!resolver.parse_global_file(find_nearest_global_file(cwd)))
+    pal::string_t global_file_path = find_nearest_global_file(cwd);
+
+    if (global_file_path.empty())
     {
-        // Fall back to a default SDK resolver
-        resolver = sdk_resolver{ allow_prerelease };
+        resolver.global_json.state = global_file_info::state::not_found;
+    }
+    else
+    {
+        global_file_info global_file = resolver.parse_global_file(global_file_path);
+        if (global_file.state != global_file_info::state::valid)
+        {
+            // Fall back to a default SDK resolver
+            resolver = sdk_resolver{ allow_prerelease };
 
-        trace::warning(
-            _X("Ignoring SDK settings in global.json: the latest installed .NET SDK (%s prereleases) will be used"),
-            resolver.allow_prerelease ? _X("including") : _X("excluding"));
+            trace::verbose(_X("Invalid global.json [%s]:\n  %s"), global_file.path.c_str(), global_file.error_message.c_str());
+            trace::warning(
+                _X("Ignoring SDK settings in global.json: the latest installed .NET SDK (%s prereleases) will be used"),
+                resolver.allow_prerelease ? _X("including") : _X("excluding"));
+        }
+
+        resolver.global_json = std::move(global_file);
```

Comment:
This is unintuitive to me. We replace `resolver` above, but still set the `global_file` on the replaced version? Can I assume the `global_json` file is checked for validity in future locations?